apps/browser-extension-wallet/src/lib/scripts/background/services/userIdService.ts

```diff
@@ -21,7 +29,30 @@ export class UserIdService implements UserIdServiceInterface {
     private sessionLength: number = SESSION_LENGTH
   ) {}
 
-  async getId(): Promise<string> {
+  async getwalletBasedUserId(networkMagic: Wallet.Cardano.NetworkMagic): Promise<string | undefined> {
+    const { keyAgentsByChain, usePersistentUserId } = await this.getStorage();
+
+    if (!keyAgentsByChain) {
+      console.debug('[ANALYTICS] Key agents not found - Wallet not created yet');
+      return this.walletBasedUserId;
```

Comment:
Could we explicitly return undefined instead (also in line 42)? 
```suggestion
      return undefined;
```